numpy/core/setup_common.py

```diff
@@ -370,32 +370,49 @@ def long_double_representation(lines):
                 # "before" bytes. In other words the first 4 "before" bytes went
                 # past the sliding window.
                 if read[:12] == _BEFORE_SEQ[4:]:
-                    if read[12:-8] == _INTEL_EXTENDED_12B:
-                        return 'INTEL_EXTENDED_12_BYTES_LE'
-                    if read[12:-8] == _MOTOROLA_EXTENDED_12B:
-                        return 'MOTOROLA_EXTENDED_12_BYTES_BE'
+                    return _long_double_representation_case1(read)
+
                 # if the content was 16 bytes, we are left with 32-8-16 = 16
                 # "before" bytes, so 8 went past the sliding window.
                 elif read[:8] == _BEFORE_SEQ[8:]:
-                    if read[8:-8] == _INTEL_EXTENDED_16B:
-                        return 'INTEL_EXTENDED_16_BYTES_LE'
-                    elif read[8:-8] == _IEEE_QUAD_PREC_BE:
-                        return 'IEEE_QUAD_BE'
-                    elif read[8:-8] == _IEEE_QUAD_PREC_LE:
-                        return 'IEEE_QUAD_LE'
-                    elif read[8:-8] == _IBM_DOUBLE_DOUBLE_LE:
-                        return 'IBM_DOUBLE_DOUBLE_LE'
-                    elif read[8:-8] == _IBM_DOUBLE_DOUBLE_BE:
-                        return 'IBM_DOUBLE_DOUBLE_BE'
+                    return _long_double_representation_case2(read)
```

Comment:
This won't work since it will return regardless of whether _long_double_representation_case2 matches any of its cases or not. Consider the case where none of the `elif` statements in _long_double_representation_case2 are matched.

Reply:
_long_double_representation_case2 would have to include a:
```
else:
    return None
```

And here there would have to be something like:

```
return_value = _long_double_representation_case2(read)
if return value:
    return return_value
```

But im not sure if this vill reduce the total CCN enough so further refactoring might be required.